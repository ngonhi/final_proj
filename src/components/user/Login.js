import React, { Component } from 'react'

class Login extends Component {
  constructor() {
    super()
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  // Handle submit event when user logs in
  handleSubmit(event) {
    event.preventDefault()
    const {username, password} = event.target.elements
    const user = {
        "username": username.value,
        "password": password.value
    }

    const url = window.$domain + '/login'
    const option = {
      method: 'POST',
      headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
      },
      body: JSON.stringify(user)
    }

    this.props.fetchRequestObj("START_LOADING_TOKEN", url, option)
    .then(() => {
      if(Object.keys(this.props.error).length !== 0) {
        console.log(this.props.error)
        this.props.history.push('/Login')
      } else {
      this.props.history.push('/Categories')
      }
    })
    //event.target.reset(); // To clear form content
  }

  componentDidUpdate() {
    if (Object.keys(this.props.user).length === 0) {
    const url = window.$domain + '/me' 
      const option = {
        method: 'GET',
        headers: {
          'Authorization': 'Bearer ' + this.props.access_token,
          'Content-Type': 'application/json'
        }
      }
    this.props.fetchRequestObj("START_LOADING_USER", url, option)
    }
  }
  
  render() {
    let error
    if (this.props.error) {
      
    if (Object.keys(this.props.error).length !== 0) {
      const {message, status, statusText} = this.props.error
      error = <div className='error'> {status} - {statusText} - {message} </div>
    } else {
      error = null
    }
  }
    return (
      <div>
        <div className='form'>
                <form onSubmit={this.handleSubmit}> 
                    <input type='text' placeholder='username' name='username'></input>
                    <input type='text' placeholder='password' name='password'></input>
                    <button> Login </button>
                </form>
        </div>
        {error}
      </div>
    )
  }
}

export default Login

{/* <div> {this.props.error ? this.props.error.map((value, index) =>
          {return <center><li key={index}>{value}</li></center> }) : null} </div> */}