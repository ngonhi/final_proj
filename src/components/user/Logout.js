import React, {Component} from 'react'
import {Redirect} from 'react-router-dom'

class Logout extends Component {
    state = {navigate: false}

    constructor() {
        super()
        this.logout = this.logout.bind(this)
    }

    logout() {
        localStorage.clear()
        this.setState({navigate: true})
    }

    render() {
        console.log('Logout')
        if (this.state.navigate) {
            return <Redirect to="/" push={true}/>
        }

        return <button className='logout-button' onClick={this.logout}> Log Out </button>
    }
}

export default Logout
//https://dev.to/calier/implementing-client-side-logout-with-react-router-v4-3a84