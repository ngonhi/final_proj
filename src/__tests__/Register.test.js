import React from 'react'
import {Register} from '../components/user/index'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<Register /> shallow rendering', () => {
    it('should have proper snapshot', () => {
      const wrapper = shallow(<Register />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
})