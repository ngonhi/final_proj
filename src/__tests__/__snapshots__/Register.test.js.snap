// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> shallow rendering should have proper snapshot 1`] = `
<div>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        name="username"
        placeholder="username"
        type="text"
      />
      <input
        name="name"
        placeholder="name"
        type="text"
      />
      <input
        name="email"
        placeholder="email"
        type="text"
      />
      <input
        name="password"
        placeholder="password"
        type="text"
      />
      <button>
         Register 
      </button>
    </form>
  </div>
  <div>
     
     
  </div>
</div>
`;
